src/rest_vol_dataset.c

```diff
@@ -3724,8 +4049,120 @@ RV_convert_buffer_to_obj_refs(char *ref_buf, size_t ref_buf_len, rv_obj_ref_t **
 static herr_t
 dataset_read_scatter_op(const void **src_buf, size_t *src_buf_bytes_used, void *op_data)
 {
-    *src_buf            = response_buffer.buffer;
-    *src_buf_bytes_used = *((size_t *)op_data);
+    response_read_info *resp_info = (response_read_info *)op_data;
+    *src_buf                      = resp_info->response_buf->buffer;
+    *src_buf_bytes_used           = *((size_t *)resp_info->read_size);
 
     return 0;
 } /* end dataset_read_scatter_op() */
+
+/* Return the index of the curl handle whose URL matches that of the given handle*/
+herr_t
+RV_get_index_of_matching_handle(CURL **curl_handles, size_t count, CURL *handle, size_t *handle_index)
```

Comment:
What's the significance of needing to find a handle with a matching URL here? Since we're searching for a `CURL *` in an array of `CURL *`, can't you just compare the pointers for equality and then return the index? Or if not, create an array of structures where the structure holds a `CURL *` and any other information you need for quick matching later? I imagine this function could get pretty slow with even a smaller number of requests due to a potential worst-case of `count` string comparisons each time we need to find the index for a handle, which also appears to happen pretty often.